quill-sql/src/main/scala/io/getquill/context/sql/norm/SqlNormalize.scala

```diff
@@ -7,13 +7,26 @@ import io.getquill.norm.RenameProperties
 
 object SqlNormalize {
 
+  private val debugEnabled = false
+
   private val normalize =
     (identity[Ast] _)
+      .andThen(debug("original"))
       .andThen(FlattenOptionOperation.apply _)
+      .andThen(debug("FlattenOptionOperation"))
       .andThen(Normalize.apply _)
+      .andThen(debug("Normalize"))
       .andThen(RenameProperties.apply _)
+      .andThen(debug("RenameProperties"))
       .andThen(ExpandJoin.apply _)
+      .andThen(debug("ExpandJoin"))
       .andThen(Normalize.apply _)
+      .andThen(debug("Normalize"))
```

Comment:
Wow, great improvement! I remember the time when I was debuging by moving each step to separate var and log that to console :)

Reply:
yeah, and this doesn't cover the steps within `Normalize` yet. I'll open an issue to provide a full-fledged query compilation tracing mechanism